internal/server/event_server.go

```diff
@@ -69,4 +79,22 @@ func (s *EventServer) ListenAndServe(stopCh <-chan struct{}) {
 	} else {
 		s.logger.Info("Event server stopped")
 	}
+
+	return nil
+}
+
+func eventKeyFunc(r *http.Request) (string, error) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return "", err
```

Comment:
What effect does returning an error here have on the status code and response message?

Reply:
See https://github.com/fluxcd/notification-controller/blob/main/internal/server/event_handlers.go#L44 a 500 error would make the sender retry and we don't want this.

Reply:
A rate limited request will receive the status 429 and retryablehttp client will only retryn on 500 responses, so we should be fine.